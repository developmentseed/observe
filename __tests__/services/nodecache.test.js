/* global fetch */
/* global test */
/* global __dirname */
/* global expect */

import {
  addNodes,
  getNodes,
  getNodesForTile,
  getNodesForTiles
} from '../../app/services/nodecache'

import nodes from '../fixtures/nodes.json'

test('add and get nodes from the cache', async () => {
  // load fixture
  const tile = '0320100322313212'
  await addNodes(tile, nodes)

  const nodeIds = nodes.map(n => [ `node/${n.id}` ])
  const nodesFromCache = await getNodes(nodeIds)
  expect(nodesFromCache).toMatchSnapshot()
})

test('get nodes for a tile', async () => {
  const tile = '0320100322313212'
  await addNodes(tile, nodes)

  const nodeIds = await getNodesForTile(tile)
  expect(nodeIds).toMatchSnapshot()
})

test('get nodes for an array of tiles', async () => {
  const tiles = ['0320100322313212', '0320100322313213']
  await addNodes(tiles[0], nodes)
  await addNodes(tiles[1], nodes)

  const tileNodes = await getNodesForTiles(tiles)
  expect(tileNodes).toMatchSnapshot()
})
