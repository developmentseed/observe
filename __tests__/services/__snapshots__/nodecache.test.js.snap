// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add and get nodes from the cache 1`] = `
Object {
  "node/8548730": Object {
    "lat": "38.9005771",
    "lon": "-77.0224428",
  },
  "node/8548853": Object {
    "lat": "38.8998216",
    "lon": "-77.0219243",
  },
  "node/8548854": Object {
    "lat": "38.8998216",
    "lon": "-77.0219243",
  },
  "node/8548856": Object {
    "lat": "38.9008903",
    "lon": "-77.0219297",
  },
  "node/8548857": Object {
    "lat": "38.9021135",
    "lon": "-77.0219297",
  },
  "node/8548858": Object {
    "lat": "38.9021135",
    "lon": "-77.0219297",
  },
  "node/8548859": Object {
    "lat": "38.9028942",
    "lon": "-77.0218653",
  },
}
`;

exports[`get nodes for a tile 1`] = `
Array [
  "node/8548730",
  "node/8548853",
  "node/8548854",
  "node/8548856",
  "node/8548857",
  "node/8548858",
  "node/8548859",
]
`;

exports[`get nodes for an array of tiles 1`] = `
Object {
  "0320100322313212": Object {
    "node/8548730": Object {
      "lat": "38.9005771",
      "lon": "-77.0224428",
    },
    "node/8548853": Object {
      "lat": "38.8998216",
      "lon": "-77.0219243",
    },
    "node/8548854": Object {
      "lat": "38.8998216",
      "lon": "-77.0219243",
    },
    "node/8548856": Object {
      "lat": "38.9008903",
      "lon": "-77.0219297",
    },
    "node/8548857": Object {
      "lat": "38.9021135",
      "lon": "-77.0219297",
    },
    "node/8548858": Object {
      "lat": "38.9021135",
      "lon": "-77.0219297",
    },
    "node/8548859": Object {
      "lat": "38.9028942",
      "lon": "-77.0218653",
    },
  },
  "0320100322313213": Object {
    "node/8548730": Object {
      "lat": "38.9005771",
      "lon": "-77.0224428",
    },
    "node/8548853": Object {
      "lat": "38.8998216",
      "lon": "-77.0219243",
    },
    "node/8548854": Object {
      "lat": "38.8998216",
      "lon": "-77.0219243",
    },
    "node/8548856": Object {
      "lat": "38.9008903",
      "lon": "-77.0219297",
    },
    "node/8548857": Object {
      "lat": "38.9021135",
      "lon": "-77.0219297",
    },
    "node/8548858": Object {
      "lat": "38.9021135",
      "lon": "-77.0219297",
    },
    "node/8548859": Object {
      "lat": "38.9028942",
      "lon": "-77.0218653",
    },
  },
}
`;
